packages/inventory/src/components/InventoryItemForm/tests/index.test.tsx

```diff
@@ -0,0 +1,609 @@
+    return (
+      <select {...otherProps} onChange={(e) => onChange(e.target.value)}>
+        {children}
+      </select>
+    );
+  };
+
+  const Select = ({ children, onChange, ...otherProps }) => {
+    return (
+      <select {...otherProps} onChange={(e) => onChange(e.target.value)}>
+        {children}
+      </select>
+    );
+  };
+
+  const Option = ({ children, ...otherProps }) => {
+    return <option {...otherProps}>{children}</option>;
+  };
+
+  Select.Option = Option;
+
+  return {
+    __esModule: true,
+    ...antd,
+    DatePicker,
+    Select,
+  };
+});
+
+describe('components/InventoryItemForm', () => {
+  beforeAll(() => {
+    store.dispatch(
+      authenticateUser(
+        true,
+        {
+          email: 'bob@example.com',
+          name: 'Bobbie',
+          username: 'RobertBaratheon',
+        },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } }
+      )
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  const props = {
+    openSRPBaseURL: 'https://mg-eusm-staging.smartregister.org/opensrp/rest/',
+    cancelURL: '/inventory-items',
+    redirectURL: 'inventory-items-done',
+    products: fixtures.products as ProductCatalogue[],
+    UNICEFSections: fixtures.unicefSections,
+    donors: fixtures.donors,
+    servicePointId: '03176924-6b3c-4b74-bccd-32afcceebabd',
+    initialValues: defaultInitialValues,
+  };
+
+  it('renders without crashing', () => {
+    shallow(
+      <Router history={history}>
+        <InventoryItemForm {...props} />
+      </Router>
+    );
+  });
+
+  it('renders correctly', () => {
+    const wrapper = mount(
+      <Router history={history}>
+        <InventoryItemForm {...props} />
+      </Router>
+    );
+
+    expect(wrapper.find('Card').props()).toMatchSnapshot('card');
+    wrapper.unmount();
+  });
+
+  it('form validation works for required fields', async () => {
+    const wrapper = mount(
+      <Router history={history}>
+        <InventoryItemForm {...props} />
+      </Router>
+    );
+
+    wrapper.find('form').simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+    });
+    wrapper.update();
+
+    expect(wrapper.find('FormItemInput').at(0).prop('errors')).toEqual(['Product is required']);
+    expect(wrapper.find('FormItemInput').at(1).prop('errors')).toEqual([]);
+    expect(wrapper.find('FormItemInput').at(2).prop('errors')).toEqual([
+      'Delivery date is required',
+    ]);
+    expect(wrapper.find('FormItemInput').at(3).prop('errors')).toEqual([
+      'Accountability end date is required',
+    ]);
+    expect(wrapper.find('FormItemInput').at(4).prop('errors')).toEqual([
+      'UNICEF section is required',
+    ]);
+    expect(wrapper.find('FormItemInput').at(5).prop('errors')).toEqual([]);
+    expect(wrapper.find('FormItemInput').at(6).prop('errors')).toEqual(['PO number is required']);
```

Comment:
should we use ids to target the formItems

Reply:
@p-netm I am targeting the `FormItemInput` components. I can use ids in this scenario. I have used ids where I am targeting the `input` element when simulating changes